web/modules/custom/joinup_search/joinup_search.behat.inc

```diff
@@ -331,6 +346,37 @@ class SearchSubContext extends DrupalSubContextBase implements DrupalSubContextI
     $this->getSession()->visit($this->locatePath('/search?keys=' . $keywords));
   }
 
+  /**
+   * Press a key in the search field.
+   *
+   * Works only in Javascript-enabled browsers.
+   *
+   * @param string $key
+   *   The human readable name of the key to press.
+   *
+   * @throws \Exception
+   *   Thrown when the browser doesn't support Javascript or when the search
+   *   field is not found.
+   *
+   * @When I press :key in the search field
+   */
+  public function pressKeyInSearchField(string $key): void {
+    if (!$this->browserSupportsJavascript()) {
+      throw new \Exception('This step requires Javascript to run.');
+    }
+
+    $elements = $this->getElementsMatchingElementAlias('search bar');
+    if (!$elements) {
```

Comment:
Instead of counting and throwing exceptions, since we are anyway kinda asserting the element, why not simply replace these if clauses with 
`Assert::assertEquals(1, count($this->getElementsMatchingElementAlias($element_name)));`
`\Drupal\joinup\Traits\TraversingTrait::getElementsMatchingElementAlias` is returning an array anyway so it should be fine.

Reply:
Taken from `\JoinupSubContext::assertElementCount`

Reply:
Great suggestion, I even found some more specific asserts to use, so I ended up with this:

```
Assert::assertNotEmpty($elements, 'Could not find the search field in the page.');
Assert::assertCount(1, $elements, 'There are multiple search fields on the page.');
```